### PR TITLE
feat(Slider): update invalid state add warning state

### LINKED ISSUE #13476: Slider validation: Dev implementation
#### Acceptance Criteria

See [final design spec](https://github.com/carbon-design-system/carbon/issues/13477#issuecomment-1509129156)

[Warning state spec here](https://github.com/carbon-design-system/carbon/issues/13477#issuecomment-1512316802)

```[tasklist]
- [x] Implement design spec - changes to react component and styling for error **and warning state**
- [x] Add unit tests to cover functionality
- [x] Ensure it's documented in storybook, has storybook controls
```

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #13476

#### Changelog

**New**

- add warn, warnText and invalidText props
- add tests for warn and invalid props

**Changed**

- update invalid styling to add icon
- update storybook controls

#### Testing / Reviewing
Check slider [`invalid`](https://deploy-preview-13674--carbon-components-react.netlify.app/?path=/story/components-slider--playground&args=invalid:true) and [`warn`](https://deploy-preview-13674--carbon-components-react.netlify.app/?path=/story/components-slider--playground&args=warn:true) states match spec

<details><summary>Design spec</summary>

image
image

</details>

### CODE REVIEW

File: packages/react/src/components/Slider/Slider.tsx [hunk 1]
```diff
@@ -84,10 +85,15 @@ export interface SliderProps
...
   /**
-   * `true` to specify if the control is invalid.
+   * `Specify whether the Slider is currently invalid
    */
   invalid?: boolean;
 
+  /**
+   * Provide the text that is displayed when the Slider is in an invalid state
+   */
+  invalidText: React.ReactNode;
```
Comment (francinelucca) [inline packages/react/src/components/Slider/Slider.tsx]: should this one be optional as well? 
```suggestion
  invalidText?: React.ReactNode;
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Mostly LGTM, one minor change suggestion.
Also is it ok that when readOnly and invalid are both true for it to show up like this? I also see it in prod but just wanted to double check 
image

**Comment by alisonjoseph** [comment]
@francinelucca ah good catch, based on the other components I believe it is a bug. fixed

**Review by francinelucca — APPROVED** [review]
🚀

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- There should be `16px` padding before and after the error/warning icon. Right now the left padding seems to be 24px.
Screen Shot 2023-05-08 at 3 30 53 PM

**Review by andreancardona — APPROVED** [review]
Other than Lauren's comment looks good on the dev side - will approve

File: packages/styles/scss/components/slider/_slider.scss [hunk 2]
```diff
@@ -138,14 +140,46 @@
...
-  }
-
   .#{$prefix}--slider__thumb:focus ~ .#{$prefix}--slider__filled-track {
     background-color: $interactive;
   }
 
+  // Invalid & warn state
+  .#{$prefix}--slider-text-input.#{$prefix}--text-input--invalid,
+  .#{$prefix}--slider-text-input--warn {
+    width: 6rem;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/slider/_slider.scss:149]:
```suggestion
    width: rem(88px);
```

**Comment by alisonjoseph** [comment]
@laurenmrice I updated it, it look correct with a two digit number, but may be issues with others?

Just as a FYI on how this is built, the number input for the slider is currently set to 4rem (64px), and with an error/warning state it is now set to 5.5rem (88px). The padding won't be correct for a 1 digit number, as it doesn't resize. It also just barely cuts off a 3 digit number at 88px wide, even though the padding for the icon is correct.  Just wanted to check that was ok, I'm not sure there is an easy way around that. 

Screenshot 2023-05-08 at 2 40 44 PM
Screenshot 2023-05-08 at 2 40 53 PM
Screenshot 2023-05-08 at 2 41 08 PM
Screenshot 2023-05-08 at 2 41 16 PM

**Review by tay1orjones — APPROVED** [review]

**Comment by alisonjoseph** [comment]
@laurenmrice updated after our discussion. Input is now always left aligned, and width is 6rem when invalid/warn is true.

**Review by laurenmrice — APPROVED** [review]

**Pull request merged**